lms/static/js/ccx/view/schedule_right_container.js

```diff
@@ -0,0 +1,221 @@
+        'click #save-changes': 'saveSchedule'
+      },
+
+      initialize: function(options) {
+        this.unsavedChanges = options.unsavedChanges;
+
+        // Hidden data will be shown into form,
+        // from there user can add this data to schedule tree
+        this.chapters = this.collection.filterFormData();
+        this.resetFormFlags();
+      },
+
+      render: function() {
+        if (this.chapters) {
+          this.loadRightContainerCCXSchedule();
+        }
+
+        return this;
+      },
+
+      loadRightContainerCCXSchedule: function() {
+        this.$el.html(_.template(scheduleRightContainerTemplate) ({
+          chaptersAvailable: (this.chapters && !_.isEmpty(this.chapters)),
+          unsavedChanges: this.unsavedChanges,
+          form: _.template(scheduleFormTemplate)({
+            chapters: this.chapters,
+            selectedChapter: this.selectedChapter,
+            selectedSubsection: this.selectedSubsection,
+            enableAddUnitToScheduleButton: this.enableAddUnitToScheduleButton
+          })
+        }));
+
+        // attach date and time pickers
+        this.$el.find('.datepair .time').timepicker({
+          'showDuration': true,
+          'timeFormat': 'G:i',
+          'autoclose': true
+        });
+
+        this.$el.find('.datepair .date').datepicker({
+          'dateFormat': 'yy-mm-dd',
+          'autoclose': true
+        });
+      },
+
+
+      addUnitInScheduleTree: function(e) {
+        // add unit to schedule tree. To make contents visible to students
+        // a unit can be chapter, sequential or a vertical.
+        e.preventDefault();
+        var chapterLocation = this.$el.find("#ccx_chapter").val();
+        var sequentialLocation = this.$el.find("#ccx_sequential").val();
+        var verticalLocation = this.$el.find("#ccx_vertical").val();
+        var startDate = this.getDateTime("start");
+        var dueDate = this.getDateTime("due");
+
+        this.trigger(
+          'showUnitInScheduleTree',
+          chapterLocation,
+          sequentialLocation === 'all' ? null : sequentialLocation,
+          verticalLocation === 'all' ? null : verticalLocation,
+          startDate,
+          dueDate
+        );
+      },
+
+      showAllUnitsInScheduleTree: function(e) {
+        // add unit to schedule tree.
+        e.preventDefault();
+        this.trigger(
+          'showAllUnitsInScheduleTree'
+        );
+      },
+
+      onChapterSelect: function (e) {
+        // On chapter select populate subsection drop box.
+        var $chapterSelect = $(e.currentTarget);
+        var chapterLocation = $chapterSelect.val();
+        var $sequentialSelect = this.$el.find('#ccx_sequential');
+        if (chapterLocation !== 'none') {
+          this.selectedChapter = this.collection.findUnit(
+            this.chapters, chapterLocation
+          );
+          this.enableAddUnitToScheduleButton = true;
+          $sequentialSelect.html('');
```

Comment:
Nit: I would suggest using empty instead of setting the HTML to an empty string.

http://api.jquery.com/empty/